backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -179,7 +179,7 @@ public Endpoint getEndpoint(String tenant, UUID id) {
             loadProperties(endpoint);
             return endpoint;
         } catch (NoResultException e) {
-            return null;
+            throw new NotFoundException("Endpoint not found");
```

Comment:
This will break existing code that is probably not tested right now:
https://github.com/RedHatInsights/notifications-backend/blob/5fb0654876102b626f06ec8a1575fc43f0b16f30/backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java#L267-L273

We already return from all REST endpoints that depend on that code, so it doesn't really need to be changed.

The only possible improvement I see would consist in adding a message that says that the endpoint was not found here:
https://github.com/RedHatInsights/notifications-backend/blob/5fb0654876102b626f06ec8a1575fc43f0b16f30/backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java#L248-L250